src/net_processing.cpp

```diff
@@ -811,6 +813,8 @@ static bool BlockRequestAllowed(const CBlockIndex* pindex, const Consensus::Para
 }
 
 PeerLogicValidation::PeerLogicValidation(CConnman* connmanIn, CScheduler &scheduler) : connman(connmanIn), m_stale_tip_check_time(0) {
+    g_enable_bip61 = gArgs.GetBoolArg("-enablebip61", DEFAULT_ENABLE_BIP61);
```

Comment:
It seems strange to initialize a global inside an object's constructor. There could be multiple `PeerLogicValidation` objects, and each will re-initialize the global.

Reply:
This is true, but:
- this global is only ever read from either methods in `PeerLogic` or static functions called by peer logic methods
- its value doesn't change post startup so if its  read more than once it will always have the same value, reinitializing it a couple times isn't really a performance issue either

some other options could be:
- make it a member variable instead of a global (will have to pass it as an arg to the static functions which reference it)
- just call GetBoolArg directly (this is done in a couple other situations)

either of the above alternate options would inflate the diff quite a bit. doing it this way seemed the most compact

and it seems like a worthwhile tradeoff to be able to eliminate the `extern` declaration. thoughts?

Reply:
I agree this is strange, would also prefer initializing it somewhere outside the constructor OR making it an instance variable if possible.

Reply:
There's no good place that's local to the file but also out of the `PeerLogicValidation` constructor (i don't think i call `GetBoolArg` safely in a static initializer)

I think that making it an instance variable is the right way to go - but its going to substantially increase the size of this diff because there are a bunch of functions invoked from PeerLogicValidation that are static to net_processing but not PeerLogicValidation methods so it will change the argument signature of all of those

If people don't mind the member var i will go ahead and do that but it will substantially increase review burden for IMO not much benefit

Reply:
(here's roughly what that diff would look like: https://github.com/skeees/bitcoin/tree/net_processing-disentangle-yugediff)

Reply:
@skeees not that big, LGTM. +1 to make it an instance variable.

Reply:
I'm also into this. +1

Reply:
if you don't mind the larger review then ok! three +1s, PR updated